src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const actionSteps = [];
+  let currentState = { ...state };
+  const add = 'addProperties';
+  const del = 'removeProperties';
+  const clear = 'clear';
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can use a destructuring for `action`
```suggestion
  for (const action of actions) {
    const { type, .......... } = action;
    switch (type) {
```